NeoMathEngine/src/CPU/CpuMathEngineDnn.cpp

```diff
@@ -32,26 +32,34 @@ void CCpuMathEngine::blobMergeByDimCommon( int dimNum, const CBlobDesc* from, co
 {
 	int s[CBlobDesc::MaxDimensions];
 	to.GetDimSizes( s );
+	// Objects count to be merged
 	int objectCount = 1;
 	for(int z  = 0; z < dimNum; z++) {
 		objectCount *= s[z];
 	}
+	// Merged object size
 	int objectSize = to.BlobSize() / objectCount;
+
+	// Splitted object sizes
+	int fromObjectSizes[CBlobDesc::MaxDimensions] = { 0 };
+	for( int i = 0; i < fromCount; i++ ) {
+		int fromLimits[CBlobDesc::MaxDimensions];
+		from[i].GetDimSizes( fromLimits );
+		fromObjectSizes[i] = 1;
```

Comment:
Something's wrong here.
`CBlobDesc::MaxDimensions` is `8`
`fromCount` limit is `MaxBlobDescs` which is `32`.

`fromObjectsSizes[i]` with `i == 8` (if `fromCount == 10`) will lead to out-of-bounds access.

Check the `Dim0` version below for an example (search for `static_assert` in this file).

P.S. the same goes for the `split` too

Reply:
Thank you. Missed it. Fixed (along with extra header cleanup).